camera/auth/src/ble/eyespy_serial_char.py

```diff
@@ -37,18 +37,22 @@ def ReadValue(self, options):
         }
         """
 
-        # Prohibit reads when the camera is trying to connect
-        # Reads generate new keys and persist them, which is *bad* when trying to connect
-        if self.wifi_manager.get_state()[0] == DeviceState.CONNECTING:
-            raise InProgressException()
-
-        self.key_manager.gen_keys()
-        out_str = json.dumps({
-            "Serial": self.key_manager.get_serial(),
-            "PubKey": self.key_manager.get_public_key().public_bytes(
-                Serialization.Encoding.PEM,
-                Serialization.PublicFormat.SubjectPublicKeyInfo
-            ).decode()
-        })
-
-        return out_str.encode("ascii")
+        try:
+            # Prohibit reads when the camera is trying to connect
+            # Reads generate new keys and persist them, which is *bad* when trying to connect
+            if self.wifi_manager.wifi_state == WifiState.CONNECTING:
+                raise InProgressException()
+
+            self.key_manager.gen_keys()
+            out_str = json.dumps({
+                "Serial": self.key_manager.serial,
+                "PubKey": self.key_manager.keys.priv_key.public_key().public_bytes(
+                    Serialization.Encoding.PEM,
+                    Serialization.PublicFormat.SubjectPublicKeyInfo
+                ).decode()
+            })
+
+            return out_str.encode("ascii")
+        except Exception as exc:
+            print(exc)
```

Comment:
We're using `logger` elsewhere, maybe we should use it here and include some context when we log the exception:
```suggestion
            logger.error("Failed to read serial characteristic", exc);
```

Reply:
Yeah there's a couple extra places where I have prints. There's a couple burired in bluezManager too and the BLE stuff.